### ISSUE TITLE
Profiling timer always armed on initialization

### ISSUE THREAD

**Issue by alk** [gperftools/gperftools#409]
Originally reported on Google Code with ID 406
```
The code currently arms the profiling timer unconditionally in ProfileHandler::RegisterThread
for timer sharing detection. Unfortunately, this means the profiling timer will stay
active, even if profiling is inactive and we have shared timers in case if no other
threads get registered later.

This poses a problem for Chromium, which links the profiling code into the executable.
See http://code.google.com/p/chromium/issues/detail?id=115149
```

Reported by `mnissler@google.com` on 2012-02-21 20:50:07

**Comment by bsilver8192** [comment]
Does anybody know what the current status on this is? I would like this issue fixed, and would be happy to take over on getting it merged.

I think @saittam is the original reporter. Any chance you're still interested in working on this, and if not, do you mind if I start with what you did?

**Comment by saittam** [comment]
I'm the original reporter indeed. Unfortunately, this has lingered in the hardly-will-ever-get-to portion of my TODO list since forever. I originally got interested in this because the continuously firing signals tended to confuse gdb (which is a different bug). I've since carried the patch that's attached to the issue in my working trees, but there was nothing else that provided a strong motivation to invest time into pushing for a resolution. At this point, I can't tell whether the problem still exists, and whether the patch still applies... sorry. The original reproduction instructions should still be valid though, so if you're seeing SIGPROF firing in a single-threaded test program after calling ProfileHandler::RegisterThread, the issue is probably still valid.

My lack of spare time to spend is only going to get worse in the upcoming months, so I'd be most delighted if you were to pick this issue up and resolve it.

**Comment by bsilver8192** [comment]
Forgot to include "Fixes: #409" in my PR #762 that just got merged which fixes this... Somebody want to close it manually?

**Comment by alk** [comment]
Fixed by https://github.com/gperftools/gperftools/pull/762. Thanks to Brian Silverman's great effort.
